docs/specs/servers/notify/notify-server-api.md

```diff
@@ -2,183 +2,244 @@
+## Webhooks
 
-**Note:** this method is idempotent and will always return the same key.
+Webhooks can be registered so your app can receive an HTTP request whenver an account subscribes or unsubscribes to notifications.
 
-`POST /subscribe-topic`
+### Webhook Request
 
-Body:
+`POST <URL>`
 
 ```typescript
-{
-  "appDomain": string,
+type Body = {
+  // JSON WebhookPayload
+  payload: string,
+  // HMAC signature of `payload` equivilent to Stripe's verification logic: https://stripe.com/docs/webhooks#verify-manually
+  signature: string,
 }
-``` 
-
-Response:
 
-```typescript
-{
-  "subscribeKey": string,
-  "authenticationKey": string,
+type WebhookPayload = {
+  // ID of the webhook that this request is for
+  webhookId: string,
+  // Unique ID of the event to deduplicate requests that were retried
+  idempotencyKey: string,
+  // Which event happened
+  event: Event,
+  // The account that triggered the event
+  account: AccountId,
+  // The new set of notification types of the account's subscription
+  scope: string[],
 }
-```
 
-```mermaid
-sequenceDiagram
-    autonumber
-    participant R as Relay
-    participant N AS Notify
-    participant D as Notify DB
-    participant A as dApp
-
-    %% Subscribe Topic
-    activate A
-    A->>+N: Call /subscribe-topic 
-    Note over N: Generate privateKeyX
-    Note over N: Generate identityKey
-    N->>+D: Store privateKeyX, identityKey
-    D-->>-N: ACK
-    N->>+R: subscribe(sha256(publicKeyX))
-    R-->>-N: ACK
-    N->>-A: did.json { publicKeyX, identityKey }
-    deactivate A
+type Event = "subscribed" | "updated" | "unsubscribed";
 ```
 
-## Register Webhook
+URL must return a 2xx status code, or the webhook request will be retried for 7 days with exponential backoff. Response body is ignored.
+
+### Register Webhook
 
-Register a webhook that will be invoked when accounts are subscribed or unsubscribed.
+Register a webhook that will be invoked when accounts subscribe to notifications, update their notification subscription, or unsubscribe.
 
-`POST /register-webhook`
+Each project is limited to a maximum of 5 webhooks.
 
-Body:
+`POST /v1/<project-id>/webhooks`
 
 ```typescript
-Webhook
+type Body = {
+  // Which events to listen for
+  events: Event[],
+  // HTTP endpoint to send Webhook Requests to
+  url: string,
+}
 ```
 
-Webhook:
-
 ```typescript
-{
-  "events": Event[], // set of Events
-  "url": string, // webhook URL
+type Response = {
+  // Webhook ID
+  id: string,
+  // Shared secret for authenticating Webhook Requests
+  secret: string,
 }
 ```
 
-Event:
+### Registered Webhooks
 
-```typescript
-"subscribed" | "unsubscribed"
-```
+Get the list of registered webhooks.
 
-Response:
+`GET /v1/<project-id>/webhooks`
 
 ```typescript
-{
-  "id": string, // webhook ID
+type Response = WebhookResponse[]
+
+type WebhookResponse = Webhook & {
+  // Webhook ID
+  id: string,
+  // Shared secret for authenticating Webhook Requests
+  secret: string,
 }
```

Comment:
Make changing the response body easier in the future w/o introducing a new endpoint version:

```suggestion
type Response = {
  webhooks: (Webhook & {
    // Webhook ID
    id: string,
    // Shared secret for authenticating Webhook Requests
    secret: string,
  })[],
}
```